x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -539,7 +583,7 @@ public UnresolvedAttribute visitDereference(EsqlBaseParser.DereferenceContext ct
     public Expression visitFunctionExpression(EsqlBaseParser.FunctionExpressionContext ctx) {
         String name = visitIdentifier(ctx.identifier());
         List<Expression> args = expressions(ctx.booleanExpression());
-        if ("count".equals(EsqlFunctionRegistry.normalizeName(name))) {
+        if (name != null && "count".equals(EsqlFunctionRegistry.normalizeName(name))) {
```

Comment:
> When can the name be null?

The check to null was add for missing function name parameter originally. For example eval `x=?fn(y)`, and `?fn` is not provided in params. After modifying `visitIdentifier` to throw a `ParsingException` for missing params, `visitFunctionExpression` should not see a null because of missing function name parameter, `visitIdentifier` may still return a null, if ctx is null, but I don't remember that I came across this code path yet.